### PR TITLE
Components

### PR DESCRIPTION
**Pull request by cheyennenoel**
Basic component; will be updated as proxy develops

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,21 +29,49 @@ jobs:
 
     steps:
       - checkout
-      - run: 
-          name: update-npm
-          command: 'sudo npm install -g npm@latest'
 
+      - run:
+          name: install mysql
```
Comment (tonykijak) [inline .circleci/config.yml]: can this be in your dependencies, installed with npm install?

File: .circleci/config.yml [hunk 2]
```diff
@@ -19,21 +29,49 @@ jobs:
...
           key: dependency-cache-{{ checksum "package.json" }}
-        
+
+      - run:
+          name: MySQL database setup
+          command: npm run create #creating database (schema.sql)
+
+      - run:
+          name: data creation
+          command: npm run seed #generating random data
```
Comment (tonykijak) [inline .circleci/config.yml]: Do you have mock data for your tests?

File: __tests__/server.test.js [hunk 3]
```diff
@@ -1,13 +1,25 @@
-// import React from 'react';
+import request from 'supertest';
 // import sinon from 'sinon';
 
+import api from '../server/start.js';
```
Comment (tonykijak) [inline __tests__/server.test.js]: Does your linter like this?

File: database/aggregate.js [hunk 4]
```diff
@@ -10,17 +10,27 @@ const generateAggregate = product => new Promise((resolve) => {
...
+    let three = 0;
+    let two = 0;
+    let one = 0;
     data.forEach((rating) => {
       total += rating.rating;
+      if (rating.rating === 5) five += 1;
+      if (rating.rating === 4) four += 1;
+      if (rating.rating === 3) three += 1;
+      if (rating.rating === 2) two += 1;
+      if (rating.rating === 1) one += 1;
```
Comment (tonykijak) [inline database/aggregate.js]: You might be able to store these ratings in an array, and use the star rating as a reference to the array's index. Increase the array value by 1.

File: src/presentational/image.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const Image = props => (
+  <div>
+    Image
+    {/*Helpful button*/}
+    {/*Not Helpful button*/}
```
Comment (tonykijak) [inline src/presentational/image.jsx]: Please don't add a button that doesn't do anything. Laurie Voss would not like that.

File: src/presentational/review.jsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    '09': 'September',
+    10: 'October',
+    11: 'November',
+    12: 'December',
+  };
+  const dates = {
+    '01': 'st',
+    '02': 'nd',
+    '03': 'rd',
+    21: 'st',
```
Comment (tonykijak) [inline src/presentational/review.jsx]: 22nd?

File: src/presentational/share.jsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import styles from '../styles.css';
+import WriteButton from './writeButton';
```
Comment (tonykijak) [inline src/presentational/share.jsx]: does your linter like this?

File: src/presentational/stars.jsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+import StarRatingComponent from 'react-star-rating-component';
```
Comment (tonykijak) [inline src/presentational/stars.jsx]: does your linter like this?

**Comment by tonykijak** [comment]
@cheyennenoel I would commit more often with smaller updates. Consider creating a separate branch for your PRs so only new information will appear.
There is a lot of commented-out code. I would try to clean this up as you go, so you don't have a broad cleanup effort at the end.

**Review by tonykijak — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (cheyennenoel) [reply .circleci/config.yml]: That's separate from npm; this is setting up my database for real-data testing on CircleCI

File: .circleci/config.yml (hunk 2 above)
Comment (cheyennenoel) [reply .circleci/config.yml]: No; will consider building mocks if time allows--especially if testing starts to run long

File: __tests__/server.test.js (hunk 3 above)
Comment (cheyennenoel) [reply __tests__/server.test.js]: Fixed

File: database/aggregate.js (hunk 4 above)
Comment (cheyennenoel) [reply database/aggregate.js]: Smart! Ticket added to Trello for optimization when time permits

File: src/presentational/image.jsx (hunk 5 above)
Comment (cheyennenoel) [reply src/presentational/image.jsx]: Pseudo for "Helpful" and "Not Helpful" review feedback buttons that will be implemented post-proxy

File: src/presentational/review.jsx (hunk 6 above)
Comment (cheyennenoel) [reply src/presentational/review.jsx]: Fixed

Comment (cheyennenoel) [reply src/presentational/review.jsx]: Good catch!

File: src/presentational/share.jsx (hunk 7 above)
Comment (cheyennenoel) [reply src/presentational/share.jsx]: Yes

File: src/presentational/stars.jsx (hunk 8 above)
Comment (cheyennenoel) [reply src/presentational/stars.jsx]: Yes

**Pull request merged**